reana_commons/gherkin_parser/parser.py

```diff
@@ -0,0 +1,258 @@
+                step_mapping[step_type][step_text] = {}
+                step_mapping[step_type][step_text]["function"] = func
+                step_mapping[step_type][step_text]["arguments"] = result.named
+                found = True
+                break
+        if not found:
+            logging.error(f"No step definition found for step: {step_text}")
+            raise StepDefinitionNotFound(
+                f"No step definition found for step: {step_text}"
+            )
+    return step_mapping
+
+
+def run_tests(
+    analysis_run_id: str,
+    workflow: str,
+    access_token,
+    feature_name: str,
+    feature_file,
+    step_mapping: Dict,
+) -> List[Dict]:
+    """Run all the tests in the parsed feature file.
+
+    :param feature_name: The name of the feature inside the feature file.
+    :param analysis_run_id:
+    :param access_token:
+    :param workflow: The name of the workflow in REANA
+    :param feature_file: The parsed and compiled feature file, as returned by `parse_feature_file`.
+    :param step_mapping: A dictionary mapping each step to the corresponding step definition (function).
+    :return: A list of dictionaries, each corresponding to one scenario of the feature file.
+    """
+    test_results = []
+    for scenario in feature_file:
+        logging.info(f"Running scenario: {scenario['name']}...")
+        result = AnalysisTestStatus.passed
+        failed_testcase = None
+        error_log = None
+        for step in scenario["steps"]:
+            logging.debug(f"Running step: {step['text']} ({step['type']})...")
+            step_type = step["type"].strip()
+            step_text = _get_step_text(step)
+            function = step_mapping[step_type].get(step_text).get("function")
+            arguments = step_mapping[step_type].get(step_text).get("arguments")
+            if function is not None:
+                try:
+                    function(workflow, access_token, **arguments)
+                except StepSkipped:
+                    logging.info(f"Scenario skipped! Current testcase: {step_text}")
+                    result = AnalysisTestStatus.skipped
+                    break
+                except Exception as e:
+                    # Catches all exceptions, including AssertionError.
+                    result = AnalysisTestStatus.failed
+                    failed_testcase = step_text
+                    error_log = str(e)
+                    logging.error(f"Scenario failed! Failed testcase: {step_text}")
+                    logging.error(f"Error log: {e}")
+                    break
+        test_results.append(
+            {
+                "scenario": scenario["name"],
+                "failed_testcase": failed_testcase,
+                "result": result,
+                "error_log": error_log,
+                "analysis_run_id": analysis_run_id,
+                "feature": feature_name,
+                "checked_at": datetime.now(),
+            }
+        )
```

Comment:
minor: maybe here we can use a proper class/dataclass instead of returning a dictionary?